backend/fms_core/template_importer/importers/sample_add_metadata.py

```diff
@@ -0,0 +1,55 @@
+from ._generic import GenericImporter
+from fms_core.template_importer.row_handlers.sample_add_metadata import SampleAddMetadataHandler
+from fms_core.templates import SAMPLE_ADD_METADATA_TEMPLATE
+
+METADATA_STARTING_INDEX = 4
+
+class SampleAddMetadataImporter(GenericImporter):
+    SHEETS_INFO = SAMPLE_ADD_METADATA_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.metadata_starting_index = METADATA_STARTING_INDEX
+
+    def import_template_inner(self):
+        """
+            METADATA SHEET
+        """
+        metadata_sheet = self.sheets['Metadata']
+
+        # Add the extra headers to the template sheet according to the metadata
+        metadata_headers = [key for i, (key, val) in enumerate(metadata_sheet.rows[0].items()) if i >= self.metadata_starting_index]
```

Comment:
makes me think of metadata name validation ... where would it go ? what would happen if a name has a space ? ... I guess this could appear in each row. We should just make sure the message is clear enough to understand it points to the header row...